### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by CBonade**
## Description
The purpose of this PR is to add a job to the end of our CI workflow and E2E workflow, that will collect up the data of all steps on a given run and store them in BigQuery. Initially, this data will be used to quantify which steps are causing the biggest hiccups in our workflow. The functionality may be expanded out over time, as we'll be storing all the step data.

### CODE REVIEW

**Review by holdenhinkle — APPROVED** [review]

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -1232,3 +1232,47 @@ jobs:
...
+      - name: Set Up BigQuery Creds
+        uses: ./.github/workflows/configure-bigquery
+
+      - name: Collect and Store Step Status
+        run: yarn actions-steps-error-check
+        working-directory: qa-standards-dashboard-data
+        env:
+          RUN_ID: ${{ github.run_id }}
+          ATTEMPT_NUMBER: ${{ github.run_attempt }}
+          MAIN_BRANCH: false
```
Comment (theodur) [inline .github/workflows/continuous-integration.yml:1278]: Not exactly sure what `MAIN_BRANCH` is for, but I'm guessing it's for keeping track of if a branch is `main`? If so, you may want to make that dynamic, since this will still run on `main` based off the `if` statement above. Maybe something like `MAIN_BRANCH: ${{ github.ref == 'refs/heads/main' }}`

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -1232,3 +1232,47 @@ jobs:
           channel_id: ${{ env.VETS_WEBSITE_CHANNEL_ID }}
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+  # Please keep steps-evaluation at the end of the workflow as it's meant to evaluate the entirety of the workflow steps that have run.
+  steps_evaluation:
+    runs-on: ubuntu-latest
+    name: Steps Evaluation
+    needs:
```
Comment (theodur) [inline .github/workflows/continuous-integration.yml:1240]: Were `unit-tests` and `contract-tests` intentionally left out?

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -1232,3 +1232,47 @@ jobs:
...
+      [
+        'build',
+        'security-audit',
+        'linting',
+        'cypress-tests',
+        'testing-reports-cypress',
+        'archive',
```
Comment (theodur) [inline .github/workflows/continuous-integration.yml:1247]: The downside of having this job need `archive` is that it'll add an extra minute to the CI workflow since the workflow will need to wait for it to finish. This will only affect branches that aren't `main` though, since `main` would run the `Deploy` job in parallel with this at the end of the workflow.

Since `archive` is relatively quick, I'm wondering if it would be better to swap `archive` with `unit-tests` since that job has more interesting metrics. This would also avoid adding an extra minute at the end of the workflow for non-`main` branches. It's not a big deal at the moment since it's only a minute being added. Just something to be cautious of if the duration of this job increases in the future

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (CBonade) [reply .github/workflows/continuous-integration.yml:1278]: This was more to track whether this comes from the CI workflow vs the full E2E run workflow. More or less, we wanted to distinguish results that could be someone just beating up a test branch from the full run. I can probably add a distinction to label by workflow, and change the `MAIN_BRANCH` to how you have it here as your explanation definitely makes more sense.

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (CBonade) [reply .github/workflows/continuous-integration.yml:1240]: I only left them out as I'm trying to ensure that it runs at the end, and cypress tests traditionally have taken longer to finish. That said, I can add them for security just in case there's an edge case where test selection makes cypress too quick.

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (CBonade) [reply .github/workflows/continuous-integration.yml:1247]: I totally get the concern about adding an extra minute. I think, in general, we'd like to keep track of even if the archive steps are failing to some degree. I wonder if the better solution is to have this kick off using a `workflow_dispatch` trigger after CI/E2E workflows finish. Maybe that's a refactor I can work on today.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (pjhill) [reply .github/workflows/continuous-integration.yml:1278]: This seems like a good proposed change. Any time we can build in resiliency is probably a worthwhile suggestion, thanks Theo.

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (pjhill) [reply .github/workflows/continuous-integration.yml:1247]: Curt, I like the idea of a refactor to kick off a separate workflow. That should eliminate the concern around workflow completion times.

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (pjhill) [reply .github/workflows/continuous-integration.yml:1240]: Sounds like adding them in won't be a problem, and should cover all bases just in case.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (theodur) [reply .github/workflows/continuous-integration.yml:1278]: That makes sense. I think adding a distinction to label by workflow sounds like a good idea. That would also make it expandable if you wanted to track more workflows in the future too

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (theodur) [reply .github/workflows/continuous-integration.yml:1247]: I like that approach of refactoring this into a separate workflow. That would also ensure this runs at the end of the workflow runs so you wouldn't need to check individual jobs. We also won't have to worry about the duration of this job increasing in the future since it'll be in it's own workflow

**Review by theodur — APPROVED** [review]
LGTM!
